### PR TITLE
Bug fix/cinfra 475 fix assertion during leader change and commit response on failure

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Made test for https://arangodb.atlassian.net/browse/CINFRA-475 work:

* Fixed test
* Fixed `transaction::Methods::~Methods()` running into an assertion, because Exceptions in `RocksDBTransactionState::doAbort()` were not handled (This is what CINFRA-475 was about)
* Fixed `Manager::updateTransaction` not handling exceptions from `transaction::Methods::commit()` (which lead to a failed commit not aborting the transaction) by making `commit()` (and `abort()` and `finish()`) `noexcept`

- [X] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Comment by goedderz** [comment]
Jenkins failure is unrelated.

File: arangod/Transaction/Methods.h [hunk 1]
```diff
@@ -206,20 +206,20 @@ class Methods {
...
-  Future<Result> abortAsync();
+  auto abortAsync() noexcept -> Future<Result>;
 
   /// @deprecated use async variant
-  Result finish(Result const& res);
+  auto finish(Result const& res) noexcept -> Result;
 
   /// @brief finish a transaction (commit or abort), based on the previous state
-  Future<Result> finishAsync(Result const& res);
+  auto finishAsync(Result const& res) noexcept -> Future<Result>;
```
Comment (maierlars) [inline arangod/Transaction/Methods.h:222]: do we want to add `[[nodiscard]]` to these functions?

**Review by maierlars — APPROVED** [review]

**Review by apetenchea — APPROVED** [review]

File: arangod/Transaction/Methods.cpp [hunk 2]
```diff
@@ -540,15 +540,12 @@ transaction::Methods::~Methods() {
         // regular life cycle. we want now to properly clean up and count them.
         _state->updateStatus(transaction::Status::COMMITTED);
       } else {
-        try {
-          this->abort();
-          TRI_ASSERT(_state->status() != transaction::Status::RUNNING);
```
Comment (mpoeter) [inline arangod/Transaction/Methods.cpp:545]: Do we want to keep this assertion?

**Review by mpoeter — APPROVED** [review]

File: arangod/Transaction/Methods.cpp (hunk 2 above)
Comment (goedderz) [reply arangod/Transaction/Methods.cpp:545]: We have the same assertion a few lines below. I thought that suffices - WDYT?

File: arangod/Transaction/Methods.h (hunk 1 above)
Comment (goedderz) [reply arangod/Transaction/Methods.h:222]: (the return values are discarded all over the place, so I decided to put it into a separate PR)

File: arangod/Transaction/Methods.cpp (hunk 2 above)
Comment (mpoeter) [reply arangod/Transaction/Methods.cpp:545]: Ah, I missed that in the diff. All good!

**Pull request merged**